src/App.tsx

```diff
@@ -10,18 +10,39 @@ import Navigation from 'components/Navigation';
 import Sidebar from 'components/Sidebar';
 import Wrapper from 'components/Wrapper';
 
+// Context
+import { Web3Context } from 'contexts/Web3';
+
 // Views
 import Deposit from 'views/Deposit';
 
 const App: React.FC = () => {
+  // Context
+  const {
+    onboard,
+    provider,
+    // wallet,
+    // notify
+  } = React.useContext(Web3Context);
+
   // State
   const [sidebar, setSidebar] = React.useState<boolean>(false);
+
+  async function readyToTransact() {
+    if (!provider) {
+      const walletSelected = await onboard?.walletSelect();
+      if (!walletSelected) return false;
+    }
+
+    const ready = await onboard?.walletCheck();
+    return ready;
+  }
```

Comment:
Hmm, does this check if a user is connected to a wallet? I'm getting the feeling this should be an effect.

Reply:
Apparently, it's meant to be called simply before each transaction (to make sure nothing has changed since the last transaction): https://github.com/pokt-network/wpokt-app/pull/4#discussion_r595238881

Although, now that I think about it, maybe the whole `readyToTransact` function should just be moved into `utils`, then called before each transaction?

Reply:
Actually, I think I see what you're saying... Lemme make a quick commit

Reply:
Fixed in 910b900a7fa76c963e13cff322c250ca782c7b16